addons/stock/models/stock_quant.py

```diff
@@ -0,0 +1,497 @@
+        readonly=True, store=True)
+    reservation_id = fields.Many2one(
+        'stock.move', 'Reserved for Move',
+        readonly=True, select=True,
+        help="The move the quant is reserved for")
+    lot_id = fields.Many2one(
+        'stock.production.lot', 'Lot',
+        ondelete="restrict", readonly=True, select=True)
+    cost = fields.Float('Unit Cost')
+    owner_id = fields.Many2one(
+        'res.partner', 'Owner',
+        readonly=True, select=True,
+        help="This is the owner of the quant")
+    create_date = fields.Datetime('Creation Date', readonly=True)
+    in_date = fields.Datetime('Incoming Date', readonly=True, select=True)
+    history_ids = fields.Many2many(
+        'stock.move', 'stock_quant_move_rel', 'quant_id', 'move_id',
+        string='Moves', copy=False,
+        help='Moves that operate(d) on this quant')
+    company_id = fields.Many2one(
+        'res.company', 'Company',
+        readonly=True, required=True, select=True,
+        default=lambda self: self.env['res.company']._company_default_get('stock.quant'),
+        help="The company to which the quants belong")
+    inventory_value = fields.Float('Inventory Value', copute='_compute_inventory_value', readonly=True)
+    # Used for negative quants to reconcile after compensated by a new positive one
+    propagated_from_id = fields.Many2one(
+        'stock.quant', 'Linked Quant',
+        readonly=True, select=True,
+        help='The negative quant this is coming from')
+    negative_move_id = fields.Many2one(
+        'stock.move', 'Move Negative Quant',
+        readonly=True,
+        help='If this is a negative quant, this will be the move that caused this negative quant.')
+    negative_dest_location_id = fields.Many2one(
+        'stock.location', "Negative Destination Location", related='negative_move_id.location_dest_id',
+        readonly=True,
+        help="Technical field used to record the destination location of a move that created a negative quant")
+
+    @api.one
+    def _compute_name(self):
+        """ Forms complete name of location from parent location to child location. """
+        self.name = '%s: %s%s' % (self.lot_id.name or self.product_id.code or '', self.qty, self.product_id.uom_id.name)
+
+    @api.multi
+    def _compute_inventory_value(self):
+        for quant in self:
+            # TDE note: there was something about poping force_company -> usefull ?
+            if quant.company_id != self.env.user.company_id:
+                # if the company of the quant is different than the current user company, force the company in the context
+                # then re-do a browse to read the property fields for the good company.
+                quant = quant.with_context(force_company=quant.company__id.id)
+            quant.inventory_value = quant._get_inventory_value()
+
+    @api.multi
+    def _get_inventory_value(self):
+        self.ensure_one()
+        return self.product_id.standard_price * self.qty
```

Comment:
The `ensure_one` call may not be necessary here, that's the second thing field access does. The one thing it will catch is calling the method on an empty recordset.